scilla/src/backend_collector.rs

```diff
@@ -0,0 +1,212 @@
+use std::collections::HashMap;
+
+use evm::backend::Backend;
+use evm_ds::protos::evm_proto::{Apply, EvmResult, Storage};
+use primitive_types::{H160, H256, U256};
+use sha3::{Digest, Keccak256};
+use tracing::{*};
+
+pub type Address = H160;
+
+#[derive(Debug, Clone)]
```

Comment:
This has been copied from the main repo to avoid the scilla library depending on the zq2 repo, thus creating a circular dependency.